python/cuml/dask/preprocessing/encoders.py

```diff
@@ -0,0 +1,357 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import cudf
+import dask
+import cupy as cp
+import numpy as np
+from cudf import Series, DataFrame
+from cudf.core import GenericIndex
+from cuml.dask.common.part_utils import _extract_partitions
+from cuml.utils import with_cupy_rmm
+from distributed import default_client
+from toolz import first
+
+
+class OneHotEncoder:
+    """
+    Encode categorical features as a one-hot numeric array.
+    The input to this transformer should be a dask_cuDF.DataFrame of ints or
+    strings, denoting the values taken on by categorical (discrete) features.
+    The features are encoded using a one-hot (aka 'one-of-K' or 'dummy')
+    encoding scheme. This creates a binary column for each category and
+    returns a sparse matrix or dense array (depending on the ``sparse``
+    parameter).
+    By default, the encoder derives the categories based on the unique values
+    in each feature. Alternatively, you can also specify the `categories`
+    manually.
+
+    Parameters
+    ----------
+    categories : 'auto' or a dask_cudf.DataFrame, default='auto'
```

Comment:
Is there a reason only DataFrame is supported? The one-hot encoder should be supporting numeric values in addition to strings, which means it should be possible to accept dask arrays here.

Reply:
Good point. This was fixed in the single GPU version. This fix was inherited in the multi-GPU version with your suggestion to use `DelayedTransformMixin` and to serialize multi-GPU models as single GPU models.